snapcraft/storeapi/__init__.py

```diff
@@ -280,15 +335,20 @@ def __init__(self, conf):
             'UBUNTU_STORE_API_ROOT_URL',
             constants.UBUNTU_STORE_API_ROOT_URL))
 
+    def get_macaroon(self, acls):
+        response = self.post(
+            'acl/',
+            data=json.dumps({'permissions': acls}),
+            headers={'Content-Type': 'application/json'})
```

Comment:
Better if you also send    Accept: application/json   as it may be used in the future (same in other requests)